### PR TITLE
Increase test coverage of Guards to 100%.

### PR DESCRIPTION
**Pull request by dhavlik**

### CODE REVIEW

**Comment by dhavlik** [comment]
travis-ci errors will be fixed by another pull request by icemac.

File: tests/test_Guards.py [hunk 1]
```diff
@@ -35,7 +36,114 @@ def test_Guards__safe_builtins__2(e_exec):
         __name__='restricted_module',
         __metaclass__=type,
         _write_=lambda x: x,
-        _getattr_=getattr)
+        _getattr_=getattr,)
```
Comment (icemac) [inline tests/test_Guards.py]: This one is not necessary.

File: tests/test_Guards.py [hunk 2]
```diff
@@ -35,7 +36,114 @@ def test_Guards__safe_builtins__2(e_exec):
...
 
     e_exec(CLASS_SOURCE, restricted_globals)
     assert restricted_globals['b'] == '2411'
+
+
+class D:
```
Comment (icemac) [inline tests/test_Guards.py]: The classes should have more meaningful values. If the are only needed for one test they could be moved into the test function together with the code which uses them.

File: tests/test_Guards.py [hunk 3]
```diff
@@ -35,7 +36,114 @@ def test_Guards__safe_builtins__2(e_exec):
...
+        _getattr_=getattr,)
+
+    e_exec(SET_ATTRIBUTE, restricted_globals)
+    assert 9999 == restricted_globals['d1'].value
+
+    e_exec(DEL_ATTRIBUTE, restricted_globals)
+    assert None is restricted_globals['d1'].value
+
+
+class E:
```
Comment (icemac) [inline tests/test_Guards.py]: Ditto.

File: tests/test_Guards.py [hunk 4]
```diff
@@ -35,7 +36,114 @@ def test_Guards__safe_builtins__2(e_exec):
...
+        __metaclass__=type,
+        _write_=lambda x: x,
+        _getattr_=getattr,)
+
+    with pytest.raises(TypeError) as excinfo:
+        e_exec(SET_ATTRIBUTE_WITHOUT_GUARDED_WRITES, restricted_globals)
+    assert 'attribute-less object (assign or del)' in str(excinfo.value)
+
+
+class F:
```
Comment (icemac) [inline tests/test_Guards.py]: Ditto.

File: tests/test_Guards.py [hunk 5]
```diff
@@ -35,7 +36,114 @@ def test_Guards__safe_builtins__2(e_exec):
...
+DEL_ATTRIBUTE = '''
+delattr(d1, 'value')
+'''
+
+
+@pytest.mark.parametrize(*e_exec)
+def test_Guards__safe_builtins__3(e_exec):
+    """It allows use setattr and delattr when _guarded_writes is True.
+
+    `__build_class__` is only needed in Python 3.
```
Comment (icemac) [inline tests/test_Guards.py]: I do not see any `__build_class__` usage in this test.

File: tests/test_Guards.py [hunk 6]
```diff
@@ -35,7 +36,114 @@ def test_Guards__safe_builtins__2(e_exec):
...
+'''
+
+
+DEL_ATTRIBUTE = '''
+delattr(d1, 'value')
+'''
+
+
+@pytest.mark.parametrize(*e_exec)
+def test_Guards__safe_builtins__3(e_exec):
```
Comment (icemac) [inline tests/test_Guards.py]: I think it should be called `test_Guards__guarded_setattr__1` as it actually tests `guarded_setattr`.

File: tests/test_Guards.py [hunk 7]
```diff
@@ -35,7 +36,114 @@ def test_Guards__safe_builtins__2(e_exec):
...
+    """It allows use setattr and delattr when _guarded_writes is True.
```
Comment (icemac) [inline tests/test_Guards.py]: The `delattr` test could be moved to another test function.

File: tests/test_Guards.py [hunk 8]
```diff
@@ -35,7 +36,114 @@ def test_Guards__safe_builtins__2(e_exec):
...
+
+SET_ATTRIBUTE_WITHOUT_GUARDED_WRITES = '''
+e1 = E()
+setattr(e1, 'foo', 'bar')
+'''
+
+
+@pytest.mark.parametrize(*e_exec)
+def test_Guards__write_wrapper__1(e_exec):
+    """It wraps it when it is not marked with _guarded_writes."""
```
Comment (icemac) [inline tests/test_Guards.py]: "`It wraps it`" ... what is `it`?

File: tests/test_Guards.py [hunk 9]
```diff
@@ -35,7 +36,114 @@ def test_Guards__safe_builtins__2(e_exec):
...
+
+SET_ATTRIBUTE_USING_GUARDED_SETATTR = '''
+f1 = F()
+setattr(f1, 'foo', 'bar')
+'''
+
+
+@pytest.mark.parametrize(*e_exec)
+def test_Guards__write_wrapper__2(e_exec):
+    """It wraps it and it works when guarded_setattr is implemented."""
```
Comment (icemac) [inline tests/test_Guards.py]: Ditto.

File: tests/test_Guards.py [hunk 10]
```diff
@@ -35,7 +36,114 @@ def test_Guards__safe_builtins__2(e_exec):
...
+        _getattr_=getattr,)
+
+    e_exec(SET_ATTRIBUTE_USING_GUARDED_SETATTR, restricted_globals)
+    assert restricted_globals['f1'].foo == 'bar'
+
+
+@pytest.mark.parametrize(*e_exec)
+def test_Guards__guarded_unpack_sequence__1(e_exec, mocker):
+    """It does not protect unpacking when the sequence is shorter
+    than expected."""
```
Comment (icemac) [inline tests/test_Guards.py]: The comment should explain why this is okay.

**Pull request merged**